menas/src/main/scala/za/co/absa/enceladus/menas/models/rest/ResponseError.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.models.rest
+
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.{JsonSubTypes, JsonTypeInfo}
+import za.co.absa.enceladus.menas.models.rest.errors.{SchemaFormatError, SchemaParsingError}
+
+/**
+  * This abstract class is used as a parent for all REST errors.
+  */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "_t")
+@JsonSubTypes(Array(
+  new Type(value = classOf[SchemaFormatError], name = "SchemaFormatError"),
+  new Type(value = classOf[SchemaParsingError], name = "SchemaParsingError")
+))
+abstract class ResponseError {
+  /** An error type. Should be as concise as possible to be parsable by a service. */
+  val errorType: String
```

Comment:
I have a feeling that this is duplicating the `_t` field Jackson adds. Maybe it's not necessary. 
1. You can change the `name` parameter in the `@JsonSubTypes` annotation to be `schema_format` and `schema_parsing`, rather than the class names.
2. You can change the `@JsonTypeInfo` annotation's `property` param to be `errorType` rather than `_t`.

You would lose this information wen deserialized to a Scala object but you have the class type available, which should be enough. 

You could also add a `serDe` test (look at the tests in `za.co.absa.enceladus.model.conformanceRule`)

Reply:
I imagined `errorType` to be more specific than the class used for storing error details. It is not used like this yet, but for the future might be useful.
Let's say we have `AuthenticationError` class that combines errors related to authentication. In this case, `errorType` can be `ldap_server_unreachable`, `kdc_server_timeout`, `invalid_keyfile`, etc. There might be much more causes than classes.